packages/gatsby-plugin-manifest/src/gatsby-ssr.js

```diff
@@ -1,25 +1,25 @@
 import React from "react"
 import { withPrefix } from "gatsby"
+import { defaultIcons } from "./common.js"
 
 exports.onRenderBody = ({ setHeadComponents }, pluginOptions) => {
+  const icons = pluginOptions.icons || defaultIcons
+
   // If icons were generated, also add a favicon link.
   if (pluginOptions.icon) {
-    let favicon = `/icons/icon-48x48.png`
+    let favicon = icons && icons.length ? `/icons/icon-48x48.png` : null
```

Comment:
So if I'm reading this right it's checking if `icons && icons.length` are truthy and if they are it's setting the favicon to be the 48px icon. This would allow for a bug in the "hybrid" mode. A user doesn't necessarily have to ask the plugin to generate a 48px icon in hybrid mode. Meaning this would point to a non existent icon. It needs to check if `icons` contains a 48px icon and either not create the favicon or add a 48px icon to be generated.

Reply:
ops - it should be `icons[0]`

Reply:
Optionally you could set this to `icons && icon ? '/icons/icon-48x48.png' : null` and just not supply a favicon in hybrid mode. You'd also have to move loading the default icons from `common.js` to bellow the favicon section.

Reply:
also see above that icons will either be user specified (hybrid or manual) or default (automatic) so it will be still '/icons/icon-48x48.png' in automatic or whatever user specified (in hybrid) - in manual (no "source" `icon`) we still won't add favicon